lib/explorer/series.ex

```diff
@@ -2244,6 +2244,71 @@ defmodule Explorer.Series do
   @spec pow(left :: Series.t() | number(), right :: Series.t() | number()) :: Series.t()
   def pow(left, right), do: basic_numeric_operation(:pow, left, right)
 
+  @euler_number 2.718281828459045
+
+  @doc """
+  Calculates the natural logarithm.
+
+  The resultant series is going to be of dtype `:float`.
+  See `log/2` for passing a custom base. 
+
+  ## Supported dtypes
+
+    * `:integer`
+    * `:float`
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list([1, 2, 3, nil, 4])
+      iex> Explorer.Series.log(s)
+      #Explorer.Series<
+        Polars[5]
+        float [0.0, 0.6931471805599453, 1.0986122886681098, nil, 1.3862943611198906]
+      >
+
+  """
+  @doc type: :element_wise
+  @spec log(argument :: Series.t()) :: Series.t()
+  def log(argument), do: log(argument, @euler_number)
+
+  @doc """
+  Calculates the logarithm on a given base.
+
+  The resultant series is going to be of dtype `:float`.
+
+  ## Supported dtypes
+
+    * `:integer`
+    * `:float`
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list([8, 16, 32])
+      iex> Explorer.Series.log(s, 2)
+      #Explorer.Series<
+        Polars[3]
+        float [3.0, 4.0, 5.0]
+      >
+
+  """
+  @doc type: :element_wise
+  @spec log(argument :: Series.t(), base :: number()) :: Series.t()
+  def log(argument, base) when is_number(base) do
+    if base <= 0, do: raise(ArgumentError, "base must be a positive number")
```

Comment:
I think that it will be better, in general, to return negative infinity for 0 in the logarithm. However, it looks like there is no support for infinity in Explorer Series/DataFrame.

![image](https://user-images.githubusercontent.com/56299341/227550526-6e40597d-4141-4ae3-b7ef-260ce16b70f8.png)

Reply:
@msluszniak we do support it, it seems the error is elsewhere. Likely Kino.Explorer.

Reply:
Yes, probably the aggregate functions like min, max, and mean don't work properly. I'll send an issue to Kino.Explorer.

Reply:
The bug is in Explorer as well. Reported here: https://github.com/elixir-nx/explorer/issues/550